### PR TITLE
changes for argocd integration

### PR DESCRIPTION
**Pull request by patilpankaj212**
1. modify Dockerfile to add a folder .ssh in /home/terrascan. This folder will be used to place ssh keys, ssh config and known_hosts required for api server to download repositories.
2. modify the resources skipping for kubernetes
3. modify response code of api server's remote repo scan response, when the result contains admission controller denied violations.

### CODE REVIEW

File: pkg/iac-providers/kubernetes/v1/normalize.go [hunk 1]
```diff
@@ -159,47 +159,16 @@ func readSkipRulesFromAnnotations(annotations map[string]interface{}, resourceID
...
-	// get rule, if rule not found return nil
-	if rule, ok = m[terrascanSkipRule]; ok {
-		if _, ok = rule.(string); ok {
-			skipRule.Rule = rule.(string)
-		} else {
+	if rules, ok := skipRulesFromAnnotations.(string); ok {
+		skipRules := make([]output.SkipRule, 0)
+		err := json.Unmarshal([]byte(rules), &skipRules)
+		if err != nil {
+			zap.S().Debugf("terrascanSkip rules '%s' cannot be unmarshalled to json", rules)
```
Comment (dev-gaur) [inline pkg/iac-providers/kubernetes/v1/normalize.go:166]:
```suggestion
			zap.S().Debugf("json string %s cannot be unmarshalled to [ ]output.SkipRules struct schema", rules)
```

**Review by dev-gaur — CHANGES_REQUESTED** [review]
One very minor change requested.

File: build/Dockerfile [hunk 2]
```diff
@@ -27,6 +27,10 @@ RUN addgroup --gid 101 terrascan && \
     adduser -S --uid 101 --ingroup terrascan terrascan && \
     apk add --no-cache git openssh
 
+# create .ssh folder and change owner to terrascan
+RUN mkdir -p /home/terrascan/.ssh && \
+    chown -R terrascan:terrascan /home/terrascan/.ssh
+
```
Comment (jlk) [inline build/Dockerfile:33]: That needs to be mode 700 as well. umask might already be set, but keep in mind...

File: pkg/iac-providers/kubernetes/v1/normalize.go (hunk 1 above)
Comment (jlk) [reply pkg/iac-providers/kubernetes/v1/normalize.go:166]: Devang's comment is technically accurate, but not sure if that much detail would be useful to average user, or confusing?

**Review by jlk — DISMISSED** [review]
Generally LGTM. Note the 2 comments, but not requiring any changes...

File: pkg/iac-providers/kubernetes/v1/normalize.go (hunk 1 above)
Comment (amirbenv) [reply pkg/iac-providers/kubernetes/v1/normalize.go]: I agree with @jlk. This change will not be informative to most users.

**Review by jlk — APPROVED** [review]
Ship it! :)

File: pkg/iac-providers/kubernetes/v1/normalize.go (hunk 1 above)
Comment (dev-gaur) [reply pkg/iac-providers/kubernetes/v1/normalize.go]: Debug is also used by us developers.. to diagnose what went wrong. Will ease it for us to help people reporting bugs and other issues.

**Review by dev-gaur — APPROVED** [review]

**Pull request merged**